sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1518,6 +1520,10 @@ class GpuRunningWindowIterator(
       fixerIndexMap.values.foreach(_.close())
       saveLastParts(Array.empty)
       saveLastOrder(Array.empty)
+      maybeSplitIter.foreach(_.close())
```

Comment:
Why loop through the iterator? It should handle closing anything that is cached itself. But that assumes that a TaskContext exists to do it. This is going to potentially cause a lot of performance problems if window is followed by a limit.

Reply:
Nice catch.
My goal is just to close the cached objects in this split iterator. Updated to call its `close` method.